### PR TITLE
New Spock

### PR DESCRIPTION
**Pull request by kasobol-msft**
Update Spock. (So we can parallelize some day).

### CODE REVIEW

File: sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/APISpec.groovy [hunk 1]
```diff
@@ -171,7 +171,7 @@ class APISpec extends Specification {
     }
 
     def setup() {
-        String fullTestName = specificationContext.getCurrentIteration().getName().replace(' ', '').toLowerCase()
+        String fullTestName = refactorName(specificationContext.getCurrentIteration().getName())
```
Comment (alzimmermsft) [inline sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/APISpec.groovy:174]: This naming is slightly off. I took a look at this in the past and this is what I ended up using: https://github.com/alzimmermsft/azure-sdk-for-java/blob/AzStorage_UpgradeSpockCore/sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/APISpec.groovy#L172

File: sdk/storage/microsoft-azure-storage-blob/pom.xml [hunk 2]
```diff
@@ -76,7 +76,7 @@
         <dependency>
             <groupId>org.spockframework</groupId>
             <artifactId>spock-core</artifactId>
-            <version>1.3-groovy-2.5</version> <!-- {x-version-update;org.spockframework:spock-core;external_dependency} -->
+            <version>2.0-M4-groovy-2.5</version> <!-- {x-version-update;org.spockframework:spock-core;external_dependency} -->
```
Comment (alzimmermsft) [inline sdk/storage/microsoft-azure-storage-blob/pom.xml:79]: Let's make an exception for Track 1 Blobs to continue using the old version of Spock

File: sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/APISpec.groovy (hunk 1 above)
Comment (kasobol-msft) [reply sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/APISpec.groovy:174]: I'll clean up name soon.

**Review by gapra-msft — APPROVED** [review]

**Comment by kasobol-msft** [comment]
I'll kick off live test run as well when I get CI in order.

**Pull request merged**